peer/configuration.proto

```diff
@@ -35,6 +35,25 @@ message ACLs {
 
 // PrivateDataImplicitCollection stores the configuration for org-specific implicit collections
 message PrivateDataImplicitCollection {
-    int32 max_peer_count = 1;
-    int32 required_peer_count = 2;
+    // The minimum number of peers private data will be sent to upon
+    // endorsement. The endorsement would fail if dissemination to at least
+    // this number of peers is not achieved.
+    int32 required_peer_count = 1;
+    // The maximum number of peers that private data will be sent to
+    // upon endorsement. This number should be greater than or equal to required_peer_count.
+    int32 maximum_peer_count = 2;
```

Comment:
I switched the order of `required_peer_count` and `max_peer_count` and also renamed `max_peer_count` to `maximum_peer_count` to be consistent with `StaticCollectionConfig`. Figured this would be the best time to do so while this proto is still unused in fabric.

Reply:
After some thought I decided to change it back to `max_peer_count` since `collection-config.json` files usually specify as `max_peer_count` instead of `maximum_peer_count`. Since the channel config is user facing `MaxPeerCount` is probably more similar to the collection config